Sources/CryptoBoringWrapper/ArbitraryPrecisionInteger.swift

```diff
@@ -381,6 +408,49 @@ extension ArbitraryPrecisionInteger: SignedNumeric {
     }
 }
 
+// MARK: - Other arithmetic operations
+
+extension ArbitraryPrecisionInteger {
+    @inlinable
+    public static func gcd(_ a: ArbitraryPrecisionInteger, _ b: ArbitraryPrecisionInteger) throws -> ArbitraryPrecisionInteger {
+        var result = ArbitraryPrecisionInteger()
+
+        guard result.withUnsafeMutableBignumPointer({ resultPtr in
+            a.withUnsafeBignumPointer { aPtr in
+                b.withUnsafeBignumPointer { bPtr in
+                    ArbitraryPrecisionInteger.withUnsafeBN_CTX { bnCtx in
+                        CCryptoBoringSSL_BN_gcd(resultPtr, aPtr, bPtr, bnCtx)
+                    }
+                }
+            }
+        }) == 1 else {
+            throw CryptoBoringWrapperError.internalBoringSSLError()
+        }
+
+        return result
+    }
+
+    @inlinable
+    public func isCoprime(with other: ArbitraryPrecisionInteger) throws -> Bool {
+        try Self.gcd(self, other) == 1
+    }
+
+    @inlinable
+    public static func random(inclusiveMin: UInt64, exclusiveMax: ArbitraryPrecisionInteger) throws -> ArbitraryPrecisionInteger {
+        var result = ArbitraryPrecisionInteger()
+
+        guard result.withUnsafeMutableBignumPointer({ resultPtr in
+            exclusiveMax.withUnsafeBignumPointer { exclusiveMaxPtr in
+                CCryptoBoringSSL_BN_rand_range_ex(resultPtr, inclusiveMin, exclusiveMaxPtr)
```

Comment:
This doesn't compile on Android armv7, a 32-bit platform, with [the following error from my Android CI](https://github.com/finagolfin/swift-android-sdk/actions/runs/9693797093/job/26750072229#step:12:444):
```
Sources/CryptoBoringWrapper/ArbitraryPrecisionInteger.swift:444:62: error: cannot convert value of type 'UInt64' to expected argument type 'BN_ULONG' (aka 'UInt32')
                CCryptoBoringSSL_BN_rand_range_ex(resultPtr, inclusiveMin, exclusiveMaxPtr)
                                                             ^
                                                             BN_ULONG(   )
```

Reply:
@finagolfin sorry about that oversight. I'm sure the fix should be pretty minimal but unfortunately I don't have a way of validating it and it's not part of our CI. Presumably the fix here is to take a `UInt` instead of a `UInt64`.

Reply:
Actually, I can reproduce this locally when building for watchOS simulator. Let me put up a patch.

Reply:
https://github.com/apple/swift-crypto/pull/242